### PR TITLE
Only notify once

### PR DESCRIPTION
**Pull request by CarvAlani**
I wanted to create an action that notifies only once instead of per every failed job

### CODE REVIEW

File: slack-check-runs-notify/src/main.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  try {
+    console.log({owner, repo, GITHUB_RUN_ID})
+
+    const jobs = (
+      await octokit.actions.listJobsForWorkflowRun({
+        owner,
+        repo,
+        run_id: GITHUB_RUN_ID,
+        per_page: 100,
```
[the remark this answers is not in the document]
Comment (CarvAlani) [reply slack-check-runs-notify/src/main.ts:44]: We're already doing this every time when checking if a job already failed 😅

File: slack-check-runs-notify/action.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  COMMIT_MESSAGE:
+    description: "COMMIT_MESSAGE"
+    required: false
+    default: ""
+  WEBHOOK_TOKEN:
+    description: "WEBHOOK_TOKEN"
+    required: true
+outputs:
+  conclusion:
+    description: "Conclusion of the job in the last workflow run for the same commit."
```
[the remark this answers is not in the document]
Comment (CarvAlani) [reply slack-check-runs-notify/action.yml:17]: No, I'll change this

File: slack-check-runs-notify/__tests__/main.test.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import {wait} from '../src/wait'
```
[the remark this answers is not in the document]
Comment (CarvAlani) [reply slack-check-runs-notify/__tests__/main.test.ts:1]: Will remove them

File: slack-check-runs-notify/src/main.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (despairblue) [reply slack-check-runs-notify/src/main.ts:42]: I know :)

File: slack-check-runs-notify/src/main.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import * as github from '@actions/github'
+import * as core from '@actions/core'
+
+const bent = require('bent')
+
+const [owner, repo] = process.env.GITHUB_REPOSITORY!.split('/', 2)
+const GITHUB_RUN_ID = Number.parseInt(process.env.GITHUB_RUN_ID!)
+
+const {
+  INPUT_GITHUB_RUN_ID: runId,
```
Comment (despairblue) [inline slack-check-runs-notify/src/main.ts:11]: Why not use `core.getInput` for all of them?

File: slack-check-runs-notify/action.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+name: 'slack-check-runs-notify'
+description: 'Check if there is a job that has failed for this commit and report it to Slack'
+author: 'Wunderflats GmbH'
+inputs:
```
Comment (despairblue) [inline slack-check-runs-notify/action.yml:4]: `GITHUB_TOKEN` is missing here? :thinking:

Comment (CarvAlani) [reply slack-check-runs-notify/action.yml:4]: It was 😅

File: slack-check-runs-notify/src/main.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    console.log({owner, repo, GITHUB_RUN_ID})
+
+    const jobs = (
+      await octokit.actions.listJobsForWorkflowRun({
+        owner,
+        repo,
+        run_id: GITHUB_RUN_ID,
+        per_page: 100,
+        // API change require this new parameter which is not yet in oktokit
+        // see: https://github.blog/changelog/2020-03-09-new-filter-parameter-in-workflow-jobs-api/
```
Comment (njibhu) [inline slack-check-runs-notify/src/main.ts:40]: This is now part of oktokit, so you can upgrade it and remove the as any

File: slack-check-runs-notify/src/main.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+const octokit: Octokit = (new github.GitHub(GITHUB_TOKEN) as any) as Octokit
+
+const runLink = `https://github.com/${owner}/${repo}/actions/runs/${runId}`
+const commit =
+  commitMessage!.trim().length > 0
+    ? `\n*${commitMessage!.trim().split('\n')[0]}*\n`
+    : ''
+
+const deploymentTestFail = {
+  text: `❌ A test check failed preventing deployment on master ${commit}<${runLink}|See github action>`
```
Comment (njibhu) [inline slack-check-runs-notify/src/main.ts:23]: We could even leave that as an input so we can use it for more than just master deployments on other repos, and maybe have a generic sentence like `A check failed for commit: <see github action>` if there's nothing set in the input

**Pull request merged**